core/src/main/java/org/apache/accumulo/core/clientImpl/ClientTabletCacheImpl.java

```diff
@@ -889,7 +889,8 @@ protected CachedTablet _findTablet(ClientContext context, Text row, boolean skip
     }
 
     if (tl == null || (locationNeed == LocationNeed.REQUIRED && tl.getTserverLocation().isEmpty()
-        && tl.getCreationTime().compareTo(cacheCutoff) < 0)) {
+        && tl.getCreationTimestamp() - cacheCutoffTimestamp < 0)) {
```

Comment:
The comparison of two `System.nanoTime()`s is the main reason these new timer objects were created in the first place so I am hoping there is a good way to refactor this to use one of the timer objects or eventually add functionality to avoid this.

Reply:
This code is trying to determine if a cache entry was created after an operation on the cache started.  This could be done w/ two Timer object and comparing their elapsed times.  If a cache entry has a smaller elapsed time than the elapsed time since the operation started then the cache entry was created after the operation stated.  If `cacheCutoff` were a Timer could do the following.

```suggestion
        && cacheCutoff.hasElapsed(tl.getCreationTimer().elapsed()))) {
```

Reply:
Could optimize the case by adding a `Timer.hasElapsed(Timer timer)` method.

Reply:
I switched back to Timer in 8e236a1. Some tests were failing with the exact syntax that you suggested even though it looks good to me.

Reply:
> Some tests were failing with the exact syntax that you suggested even though it looks good to me.

Ah, I had it backwards.   Should be `tl.getCreationTimer().hasElapsed(cacheCutoff.elapsed())` which is equivalent to the change you made.